src/glue/SessionHL.cpp

```diff
@@ -167,6 +172,33 @@ bool SessionHL::Authenticate(const std::string &username, const std::string &pas
   return res;
 }
 
+bool SessionHL::SSOAuthenticate(const std::string &scheme, const std::string &identity_provider_response) {
+  bool res = true;
+  interpreter_.ResetUser();
+  {
+    auto locked_auth = auth_->Lock();
+    if (locked_auth->AccessControlled() && FLAGS_auth_sso_on) {
```

Comment:
Put `FLAGS_auth_sso_on` inside the access controlled logic

Reply:
Done

Reply:
Update: With `FLAGS_auth_sso_on` removed (as agreed), `Auth::AccessControlled` now checks if there are external auth modules, disabling userless logins if that’s the case